alg/teca_temporal_index_select.cxx

```diff
@@ -0,0 +1,337 @@
+
+    // skip modifying the request if no indices have been requested
+    if (! (this->indices.size() == 0) )
+    {
+        // get the key for the time indices requested
+        std::string key;
+        if (request_in.get("index_request_key", key))
+        {
+            TECA_FATAL_ERROR("failed to get index request key")
+            return up_reqs;
+        }
+        std::vector<size_t> index_extent;
+        request_in.get(key, index_extent);
+
+        // create a request for each index in the request
+        for ( auto i = index_extent[0]; i <= index_extent[1]; ++i)
+        {
+            // check that this index is valid
+            if ( i >= this->indices.size() )
+            {
+                TECA_FATAL_ERROR(
+                    << "index_request_key: " << 
+                    index_extent[0] << ":" << index_extent[1]
+                    << std::endl
+                    << "Bad request: list index " << i 
+                    << " is out of bounds of the index list of size "
+                    << this->indices.size())
+            }
+
+            // copy the incoming request
+            teca_metadata req = teca_metadata(request_in);
+
+            // map the incoming request index to the list of temporal indices provided
+            std::vector<size_t> new_indices = {0,0};
+            new_indices[0] = this->indices[i];
+            new_indices[1] = this->indices[i];
+            req.set(key, new_indices);
+
+            // append this request to the list
+            up_reqs.push_back(req);
+        }
+    }
+    else
+    {
+        // just pass along the input request
+        up_reqs.push_back(request_in);
+    }
+
+    return up_reqs;
+}
+
+// --------------------------------------------------------------------------
+const_p_teca_dataset teca_temporal_index_select::execute(unsigned int port,
+    const std::vector<const_p_teca_dataset> &input_data,
+    const teca_metadata &request)
+{
+#ifdef TECA_DEBUG
+    std::cerr << teca_parallel_id() << "teca_temporal_index_select::execute" << std::endl;
+#endif
+    (void)port;
+    (void)request;
+
+    std::ostringstream oss;
+
+
+    // check for valid incoming data
+    if (input_data.size() == 0)
+    {
+        TECA_FATAL_ERROR("Failed to obtain indices; incoming dataset is empty")
+        return nullptr;
+    }
+
+    // get the input mesh
+    const_p_teca_mesh in_mesh
+        = std::dynamic_pointer_cast<const teca_mesh>(input_data[0]);
+
+    if (!in_mesh)
+    {
+        TECA_FATAL_ERROR("Failed to obtain indices; dataset is not a teca_mesh")
+        return nullptr;
+    }
+
+    // default to CPU operations
+    int device_id = -1;
+    teca_variant_array::allocator allocator = 
+        teca_variant_array::allocator::malloc;
+#if defined(TECA_HAS_CUDA)
+    request.get("device_id",device_id);
+
+    // set the CUDA device to run on
+    cudaError_t ierr = cudaSuccess;
+    if (( ierr = cudaSetDevice(device_id)) != cudaSuccess )
```

Comment:
put this in the `if (device_id >= 0)` block.

Reply:
good catch, thanks; done